client/payment-request/index.js

```diff
@@ -429,6 +438,14 @@ jQuery( ( $ ) => {
 				wcpayPaymentRequest.addToCart();
 			} );
 
+			// WooCommerce Deposits support.
+			// Trigger the "woocommerce_variation_has_changed" event when the deposit option is changed.
+			$( 'input[name=wc_deposit_option]' ).on( 'change', () => {
```

Comment:
```suggestion
			$( 'input[name=wc_deposit_option],input[name=wc_deposit_payment_plan]' ).on( 'change', () => {
```

This causes same to happen when the payment plan is changed if multiple payment plans are available.